src/commands/compile.rs

```diff
@@ -0,0 +1,280 @@
+//! The module that implements the `wasmtime compile` command.
+
+use crate::CommonOptions;
+use anyhow::{bail, Context, Result};
+use std::fs;
+use std::path::PathBuf;
+use structopt::{
+    clap::{AppSettings, ArgGroup},
+    StructOpt,
+};
+use target_lexicon::Triple;
+use wasmtime::Engine;
+
+lazy_static::lazy_static! {
+    static ref AFTER_HELP: String = {
+        format!(
+            "By default, no CPU features or presets will be enabled for the compilation.\n\
+            \n\
+            {}\
+            \n\
+            Usage examples:\n\
+            \n\
+            Compiling a WebAssembly module for the current platform:\n\
+            \n  \
+            wasmtime compile example.wasm
+            \n\
+            Specifying the output file:\n\
+            \n  \
+            wasmtime compile -o output.cwasm input.wasm\n\
+            \n\
+            Compiling for a specific platform (Linux) and CPU preset (Skylake):\n\
+            \n  \
+            wasmtime compile --target x86_64-unknown-linux --cranelift-enable skylake foo.wasm\n",
+            crate::WASM_FEATURES.as_str()
+        )
+    };
+}
+
+/// Compiles a WebAssembly module.
+#[derive(StructOpt)]
+#[structopt(
+    name = "compile",
+    version = env!("CARGO_PKG_VERSION"),
+    setting = AppSettings::ColoredHelp,
+    group = ArgGroup::with_name("x64").multiple(true),
+    group = ArgGroup::with_name("preset-x64"),
+    group = ArgGroup::with_name("aarch64").multiple(true).conflicts_with_all(&["x64", "preset-x64"]),
+    group = ArgGroup::with_name("preset-aarch64").conflicts_with_all(&["x64", "preset-x64"]),
```

Comment:
Are these groups still applicable? (I won't pretend to know what clap is doing here)

Reply:
Good eye, these should be removed.